test/inputs-dovetail-bytes-test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const support = require('./support');
+const bigquery = require('../lib/bigquery');
+const DovetailBytes = require('../lib/inputs/dovetail-bytes');
+
+describe('dovetail-bytes', () => {
+
+  let bytes = new DovetailBytes();
+
+  it('recognizes bytes records', () => {
+    expect(bytes.check({})).to.be.false;
+    expect(bytes.check({type: 'anything'})).to.be.false;
+    expect(bytes.check({type: 'bytes'})).to.be.true;
+    expect(bytes.check({type: 'segmentbytes'})).to.be.true;
+  });
+
+  it('formats download byte inserts', () => {
+    const record = bytes.formatDownload({
+      timestamp: 1490827132999,
+      request_uuid: 'the-uuid',
+      bytes_downloaded: 123,
+      seconds_downloaded: 1.23,
+      percent_downloaded: 0.12,
+    })
+    expect(record).to.have.keys('insertId', 'json');
+    expect(record.insertId).to.equal('bytes-the-uuid');
+    expect(record.json).to.have.keys('timestamp', 'request_uuid', 'bytes', 'seconds', 'percent');
+    expect(record.json.timestamp).to.equal(1490827132);
+    expect(record.json.request_uuid).to.equal('the-uuid');
```

Comment:
Does the request uuid come from kinesis?

Reply:
Yep, and it actually originates from Dovetail (the ECS app).  When it returns a 302 to the CDN, it also generates a request-uuid that gets (1) logged to kinesis, and (2) appended to the CDN url as `?reqid=1234`.